js/ui/map.js

```diff
@@ -185,10 +185,34 @@ util.extend(Map.prototype, {
     project: function(latlng) {
         return this.transform.locationPoint(LatLng.convert(latlng));
     },
+
     unproject: function(point) {
         return this.transform.pointLocation(Point.convert(point));
     },
 
+    getSize: function() {
+        return new Point(
+            this._container.clientWidth,
+            this._container.clientHeight
+        );
+    },
+
+    getPixelOrigin: function(center, zoom) {
+        var viewHalf = this.getSize().div(2);
+        return this.project(center, zoom).sub(viewHalf).add(new Point(0, 0)).round();
+    },
+
+    layerPointToLatLng: function(point) {
+        var projectedPoint = new Point(point.x, point.y)
+            .add(this.getPixelOrigin(this.getCenter(), this.getZoom()));
+        return this.unproject(projectedPoint);
+    },
+
+    containerPointToLatLng: function(point) {
+        var layerPoint = new Point(point.x, point.y).sub(new Point(0, 0));
+        return this.layerPointToLatLng(layerPoint);
+    },
+
```

Comment:
All these methods are not necessary — just use `map.unproject(point)` to get a `LatLng`.